### PR TITLE
feat(search): add highlighting

### PR DESCRIPTION
**Pull request by Haroenv**
Changes proposed in this pull request:

- adds highlighting to the search 
- adds attributes to retrieve to the search (only request what's shown, makes the search a bit faster)

some questions that I need to check before it should be merged (but the main idea can already been reviewed)

- [ ] should the highlight tag be mark, or strong
- [x] is the code correctly indented (I committed from GitHub, didn't run prettier)
- [x] does every hit have a title (aka should ?. be used)

### CODE REVIEW

**Review by MichaelSolati — APPROVED** [review]

**Comment by MichaelSolati** [comment]
@robdodson I think it's working well but may not fit the style of the site, wanna take a peek real quick?

**Comment by robdodson** [comment]
yeah I'm curious if it might match better if we use `<strong>` instead of `<mark>`

**Comment by robdodson** [comment]
thank you for working on this @Haroenv!

**Comment by francoischalifour** [comment]
We usually recommend using `mark` for highlighting so I would keep it rather than `strong` but style the tag differently—the default browser styling for `mark` is not great.

**Comment by Haroenv** [comment]
I can change it to whatever you prefer :)

**Comment by robdodson** [comment]
@Haroenv would you be cool swapping it to `<strong>`?

**Comment by robdodson** [comment]
oh sorry, I didn't see @francoischalifour's comment.

@francoischalifour why do you recommend mark over strong? is it for accessibility reasons?

We could ship a mark style, but I imagine we'd extend our strong style.

**Comment by robdodson** [comment]
btw our strong style, which is in _type.scss is:

```
strong {
  font-weight: $FONT_WEIGHT_MEDIUM;
}
```

File: src/lib/components/Search/index.js [hunk 1]
```diff
@@ -341,7 +344,13 @@ class Search extends BaseElement {
     }
     try {
       const index = await loadAlgoliaLibrary();
-      const {hits} = await index.search(query, {hitsPerPage: 10});
+      const {hits} = await index.search(query, {
+        hitsPerPage: 10,
+        attributesToHighlight: ['title'],
+        attributesToRetrieve: ['url'],
+        highlightPreTag: '<mark>',
+        highlightPostTag: '</mark>',
```
Comment (Haroenv) [inline src/lib/components/Search/index.js:352]:
```suggestion
        highlightPreTag: '<strong>',
        highlightPostTag: '</strong>',
```

going with strong here, it's consistent with google search (which uses `b`)

**Comment by Haroenv** [comment]
I've swapped it @robdodson :)

**Comment by Haroenv** [comment]
hey @robdodson! it's the way you prefer it now :)

**Comment by robdodson** [comment]
great thank you! I think I may do a follow up to make the <strong> styles a bit more noticeable

**Comment by Haroenv** [comment]
Yes, seems like the bold is quite subtle now, but nice to see it live @robdodson! Don't hesitate to reach out if you have any feedback around algolia by the way

**Comment by robdodson** [comment]
@Haroenv actually I was wondering if you have any advice on reducing the number of search requests we're sending over on developer.chrome.com. Currently we are doing an autocomplete as you type search, but I recently had to debounce it to only send every 500ms because sending with every keystroke was causing our algolia bill to become very expensive 😅

**Pull request merged**